### ISSUE TITLE
Installation reported as successfull despite error in postinstall script

### ISSUE THREAD

**Issue by lukaszfiszer** [pact-foundation/pact-node#156]
## Software versions

-   **OS**: e.g. Windows 10
-   **Library version**: e.g. `8.2.0`
-   **Node Version**: `10.15.1`

## Expected behaviour

When a postinstall scripts fails to download binaries or validate their checksum, the process should be terminated so that npm/yarn knows that it has failed, resulting in failed installation

## Actual behaviour

A warning is printed in console but the process does not exit resulting in npm install reported as successful, when in fact it was not.

## Relevant log files

```
λ npm install @pact-foundation/pact-node
// ...
> @pact-foundation/pact-node@8.2.0 postinstall C:\Users\hlukafis\projects\pact-node\node_modules\@pact-foundation\pact-node
> node postinstall.js

Installing Pact Standalone Binary for win32.
Downloading Pact Standalone Binary v1.64.1 for platform win32 from https://nexus.hltech.dev/repository/binaries/pact/pact-1.64.1-win32.zip
Finished downloading binary to C:\Users\hlukafis\projects\pact-node\node_modules\@pact-foundation\pact-node\standalone\pact-1.64.1-win32.zip
Extracting binary from C:\Users\hlukafis\projects\pact-node\node_modules\@pact-foundation\pact-node\standalone\pact-1.64.1-win32.zip.
(node:33612) UnhandledPromiseRejectionWarning: Error: Error while installing binary: Postinstalled Failed Unexpectedly: Error: Error while installing binary: Extraction failed for C:\Users\hlukafis\projects\pact-node\node_modules\@pact-foundation\pact-node\standalone\pact-1.64.1-win32.zip: Error: Error while installing binary: Checksum rejected for file 'pact-1.64.1-win32.zip' with checksum pact-1.64.1-win32.zip.checksum
    at throwError (C:\Users\hlukafis\projects\pact-node\node_modules\@pact-foundation\pact-node\standalone\install.js:232:11)
    at C:\Users\hlukafis\projects\pact-node\node_modules\@pact-foundation\pact-node\standalone\install.js:265:38
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:33612) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:33612) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN pact-node@1.0.0 No description
npm WARN pact-node@1.0.0 No repository field.

+ @pact-foundation/pact-node@8.2.0
added 8 packages from 3 contributors, removed 7 packages and updated 199 packages in 18.912s
```

**Comment by lukaszfiszer** [comment]
The fix would beto catch failed promise in [postinstall.js](https://github.com/pact-foundation/pact-node/blob/master/postinstall.js) and do `process.exit(1)` there

**Comment by TimothyJones** [comment]
Thanks for reporting this! I've fixed it in #195

**Comment by TimothyJones** [comment]
Will be released in 10.0.1.
